homeassistant/components/tuya/climate.py

```diff
@@ -37,34 +44,53 @@
 FAN_MODES = {FAN_LOW, FAN_MEDIUM, FAN_HIGH}
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up Tuya Climate devices."""
-    if discovery_info is None:
-        return
-    tuya = hass.data[DATA_TUYA]
-    dev_ids = discovery_info.get("dev_ids")
-    devices = []
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up tuya sensors dynamically through tuya discovery."""
+
+    platform = config_entry.data[CONF_PLATFORM]
+
+    async def async_discover_sensor(dev_ids):
+        """Discover and add a discovered tuya sensor."""
+        if not dev_ids:
+            return
+        entities = await hass.async_add_executor_job(
+            _setup_entities, hass, dev_ids, platform,
+        )
+        async_add_entities(entities)
+
+    async_dispatcher_connect(
+        hass, TUYA_DISCOVERY_NEW.format(SENSOR_DOMAIN), async_discover_sensor
+    )
+
+    devices_ids = hass.data[DOMAIN]["pending"].pop(SENSOR_DOMAIN)
+    await async_discover_sensor(devices_ids)
+
+
+def _setup_entities(hass, dev_ids, platform):
+    """Set up Tuya Climate device."""
+    tuya = hass.data[DOMAIN][TUYA_DATA]
+    entities = []
     for dev_id in dev_ids:
-        device = tuya.get_device_by_id(dev_id)
-        if device is None:
+        entity = tuya.get_device_by_id(dev_id)
```

Comment:
Please don't call this `entity`. That name is reserved for `Entity` instances.

Reply:
Also in case of local variable?

Reply:
Yes, always.